packages/vsx-registry/src/browser/vsx-extensions-widget.ts

```diff
@@ -57,14 +59,31 @@ export class VSXExtensionsWidget extends SourceTreeWidget {
         this.addClass('theia-vsx-extensions');
 
         this.id = generateExtensionWidgetId(this.options.id);
+
+        this.toDispose.push(this.extensionsSource);
+        this.source = this.extensionsSource;
+
         const title = this.options.title ?? this.computeTitle();
         this.title.label = title;
         this.title.caption = title;
 
-        this.toDispose.push(this.extensionsSource);
-        this.source = this.extensionsSource;
+        this.toDispose.push(this.source.onDidChange(async () => {
+            this.badge = await this.resolveCount();
+        }));
+    }
+
+    private _badgeCount: number | undefined = undefined;
```

Comment:
minor: we should rename to `_badge` due to the getter and setter, or just use `badgeCount` (dropping the unnecessary `_`). I'm in favor of using `_badge` however.

- https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#no_underscore_private

Reply:
I agree with you, `_badge` is more appropriate, 
It has been updated in the latest commit, 
I have also squashed the commits and re-based it to later master.

Thanks @vince-fugnitto !